velox/exec/Window.cpp

```diff
@@ -187,6 +193,31 @@ void Window::createWindowFunctions() {
   }
 }
 
+// The supportRowLevelStreaming is designed to support 'rank' and
+// 'row_number' functions and the agg window function with default frame.
+bool Window::supportRowLevelStreaming() {
+  for (const auto& windowNodeFunction : windowNode_->windowFunctions()) {
+    const auto& functionName = windowNodeFunction.functionCall->name();
+    auto windowFunctionEntry =
+        exec::getWindowFunctionEntry(functionName).value();
+    if (windowFunctionEntry->processingUnit == ProcessingUnit::kPartition) {
```

Comment:
Do we really need to introduce `ProcessingUnit` flag? According to this part, kRow flag does convince us that this function can be processed by rows streaming. Maybe, the checking in the following code is enough? Or make `ProcessingUnit` flag convincible, and we only need to check it.

Reply:
@PHILO-HE The `ProcessingUnit `is utilized to prevent instantiation of the `WindowFunction `object when accessing this metadata variable. Detailed discussions can be found [here](https://github.com/facebookincubator/velox/pull/9025#discussion_r1522685435).